examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,399 @@
+        return None
+
+    # Prepare topic choices for user selection
+    topic_choices = "\n".join(
+        [f"{i + 1}. {topic[0]}" for i, topic in enumerate(topics)]
+    )
+    user_input = Prompt.ask(
+        f"Select a debate topic:\n{topic_choices}",
+        choices=[str(i + 1) for i in range(len(topics))],
+        default="1",
+    )
+    topic_index = int(user_input) - 1
+
+    selected_topic = topics[topic_index]
+    logger.info(f"Selected topic: {selected_topic[0]}")
+    return selected_topic
+
+
+def select_side(topic_name: str) -> Literal["pro", "con"]:
+    """Prompt the user to select their side in the debate.
+
+    Presents the user with a choice to debate on either the pro or con side
+    of the given topic.
+
+    Args:
+        topic_name (str): The name of the debate topic.
+
+    Returns:
+        Literal["pro", "con"]: The selected debate side.
+    """
+    side = Prompt.ask(
+        f"Which side would you like to debate on?\n1. Pro-{topic_name}\n2. Con-{topic_name}",
+        choices=["1", "2"],
+        default="1",
+    )
+    return "pro" if side == "1" else "con"
+
+
+def select_topic_and_setup_side(system_messages: SystemMessages) -> Tuple[str, str, str, str]:
+    """Prompt the user to select a debate topic and sets up the respective side.
+
+    This function handles the user interaction for selecting a debate topic and the side
+    (Pro or Con) they want to argue. It validates that a topic is selected and raises an
+    exception if the topic is not available.
+
+    Args:
+        system_messages (SystemMessages): The object containing system messages with respective
+                                          debate topics.
+
+    Returns:
+        Tuple[str, str, str, str]: A tuple containing:
+            - topic_name (str): The name of the selected debate topic.
+            - pro_key (str): The key for the Pro side of the selected topic.
+            - con_key (str): The key for the Con side of the selected topic.
+            - side (str): The user's selected side, either "pro" or "con".
+
+    Raises:
+        ValueError: If no topic is selected or no topics are available in the provided
+                    `system_messages`.
+    """
+    selected_topic_tuple = select_debate_topic(system_messages)
+    if not selected_topic_tuple:
+        logger.error("No topic selected. Exiting.")
+        raise ValueError("No topic selected.")
+
+    topic_name, pro_key, con_key = selected_topic_tuple
+    side = select_side(topic_name)
+    return topic_name, pro_key, con_key, side
+
+
+def is_llm_delegate() -> bool:
+    """Prompt the user to decide on LLM delegation.
+
+    Asks the user whether the LLM should autonomously continue the debate
+    without requiring user input.
+
+    Returns:
+        bool: True if the user chooses LLM delegation, otherwise return False.
+    """
+    return Confirm.ask(
+        "Would you like the LLM to autonomously continue the debate without "
```

Comment:
this question is confusing - you're asking two questions, then y/n. Make the question crisp so that it is clear that a yes/no answer applies

Reply:
Done